### ISSUE TITLE
Add PGO support

### ISSUE THREAD

**Issue by zamazan4ik** [openobserve/openobserve#911]
### Which OpenObserve functionalities are relevant/related to the feature request?

other

### Description

Apply Profile-Guided Optimization (PGO) to OpenObserve to achieve even more performance. Could be useful to read: https://github.com/ZaMaZaN4iK/awesome-pgo

### Proposed solution

Test and/or document using PGO on the project.

### Alternatives considered

Left things as is :)

**Comment by hengfeiyang** [comment]
Interesting

**Comment by wojtekLs** [comment]
@zamazan4ik `PGO`? Did you mean `PDO` (database driver)?

**Comment by zamazan4ik** [comment]
@wojtekLs no, I mean PGO - Profile-Guided Optimization: https://en.wikipedia.org/wiki/Profile-guided_optimization

**Comment by wojtekLs** [comment]
@zamazan4ik never heard of it. Will read your link. Thanks.

**Comment by zamazan4ik** [comment]
@wojtekLs I highly recommend to read the materials from here: https://github.com/ZaMaZaN4iK/awesome-pgo Here I collected multiple articles about PGO, real-life scenarios (and results) and much more (like post-PGO optimizers like BOLT).
